dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -953,7 +953,7 @@ private void trimStringInScratchBuffer(boolean isUnicode, int[] range)
         {
             while (range[0] < range[1] - 1)
             {
-                int codePoint = this.scratchBuffer[range[0]] << 8 | this.scratchBuffer[range[0]+1];
+                int codePoint = (this.scratchBuffer[range[0]] & 0xFF) | ((this.scratchBuffer[range[0]+1] & 0xFF) << 8);
```

Comment:
@jpmcmu this needs a source code comment, let's make sure the next person working on this understands why this is being done.